config/backend-config/internal/cache/cache.go

```diff
@@ -0,0 +1,173 @@
+//go:generate mockgen -destination=./mock_cache.go -package=cache -source=./cache.go cache
+package cache
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+
+	"github.com/rudderlabs/rudder-server/config"
+	migrator "github.com/rudderlabs/rudder-server/services/sql-migrator"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+	"github.com/rudderlabs/rudder-server/utils/pubsub"
+)
+
+var pkgLogger = logger.NewLogger().Child("backend-config-cache")
+
+type Cache interface {
+	Get(ctx context.Context) ([]byte, error)
+}
+
+type cacheStore struct {
+	*sql.DB
+	secret string
+	key    string
+}
+
+// Start returns a new Cache instance, and starts a goroutine to cache the config
+//
+// secret is the secret key to encrypt the config with before storing it
+//
+// key is the key to use to store and fetch the config from the cache store
+//
+// ch is the channel to listen on for config updates and store them
+func Start(ctx context.Context, secret, key string, ch pubsub.DataChannel) Cache {
+	var (
+		err    error
+		dbConn *sql.DB
+	)
+	// setup db connection
+	dbConn, err = setupDBConn()
+	dbStore := cacheStore{
+		dbConn,
+		secret,
+		key,
+	}
+	if err != nil {
+		pkgLogger.Errorf("failed to setup db: %v", err)
+		return &dbStore
+	}
+
+	// apply migrations
+	err = migrate(dbConn)
+	if err != nil {
+		pkgLogger.Errorf("failed to migrate db: %v", err)
+		return &dbStore
+	}
+
+	// clear config for other keys
+	err = dbStore.clear(ctx)
```

Comment:
Is it enough to clear other keys during startup? Can a long running multi-tenant node create multiple different keys and entries in the cache during its lifetime?

Reply:
It can
but every time there's a workspace change
this functions is called again
so it'll cklear all the keys during the server's lifetime.
Didn't really find it necessary to have another routine that clears the table up.